cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -60,7 +60,7 @@ namespace py {
 
 using internal::NumPyTypeSize;
 
-constexpr int64_t kBinaryMemoryLimit = std::numeric_limits<int32_t>::max();
+constexpr int64_t kBinaryMemoryLimit = std::numeric_limits<int32_t>::max() - 1;
```

Comment:
Apparently this is already available as `BinaryBuilder::kMaximumCapacity` (though for some reason it's protected).

Reply:
I can move this constexpr to builder.h and use that consistently

Reply:
done